app/timetables/tests/test_models.py

```diff
@@ -65,3 +65,63 @@ def test_meal_end_time_same_with_start_time_cannot_be_saved(self):
         )
 
         self.assertRaises(ValidationError, meal.save)
+
+
+class TimetableTest(TestCase):
+    """Tests the Timetable model."""
+
+    def setUp(self):
+        Timetable.objects.create(
+            name='timtable-item',
+            code='FBI23212',
+            api_key='419223',
+            cycle_length=1,
+            current_cycle_day=2,
+            date_created=datetime.time(datetime.strptime('5:7:9', '%H:%M:%S')),
+            date_modified=datetime.time(datetime.strptime('10:7:9', '%H:%M:%S'))
+        )
+    
+    def test_timetable_name_should_be_capitalized_on_save(self):
+        timetable = Timetable.objects.get(name__iexact='timtable-item')
+        
+        self.assertEqual(timetable.name, 'Timtable-item')
+
+    def test_duplicate_timetable_name_cannot_be_saved(self):
+        timetable = Timetable(
+            name='timtable-item',
+            code='FB23212',
+            api_key='41923',
+            cycle_length=1,
+            current_cycle_day=2,
+            date_created=datetime.time(datetime.strptime('4:7:9', '%H:%M:%S')),
+            date_modified=datetime.time(datetime.strptime('15:7:9', '%H:%M:%S'))
+        )
+
+        self.assertRaises(IntegrityError, timetable.save)
+
+    def test_duplicate_timetable_code_cannot_be_saved(self):
+        timetable = Timetable(
+            name='timtable-value',
+            code='FBI23212',
+            api_key='41923',
+            cycle_length=1,
+            current_cycle_day=2,
+            date_created=datetime.time(datetime.strptime('4:7:9', '%H:%M:%S')),
+            date_modified=datetime.time(datetime.strptime('15:7:9', '%H:%M:%S'))
+        )
+        
+        self.assertRaises(IntegrityError, timetable.save)
+    
+    def test_duplicate_timetable_code_cannot_be_saved(self):
+        timetable = Timetable(
```

Comment:
This is a duplicate test as there is already this same one above @andela-joyebanji